### PR TITLE
[MUI-552]: Create Dropdown Component

### PR DESCRIPTION
**Pull request by mr-teddy-bear**
# Description

Create Dropdown Component

## Jira ticket(optional)

[MUI-552](https://minterestfinance.atlassian.net/jira/software/projects/MUI/boards/7?assignee=62591503eee0a9006ab98b74&selectedIssue=MUI-552)

## Type of change

Please pick options that are relevant.

- [ ] Bug fix OR refactoring (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature or refactoring that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+export const Dropdown = (props: Props<Option, false, any>): JSX.Element => {
+  return (
+    <Select
+      components={{ DropdownIndicator: () => <SelectArrow /> }}
+      isSearchable={false}
+      options={props.options}
+      styles={styles}
+      {...props}
+      formatOptionLabel={(asset) => (
```
Comment (gavrilin-remote) [inline src/components/Dropdown/Dropdown.tsx:95]: no need to bind naming to asset, make component more generic

```suggestion
      formatOptionLabel={(option) => (
```

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        <div className='customOption'>
+          {asset.image && (
+            <img className='customOption-img' src={asset.image} alt='asset' />
+          )}
+          <span className='customOption-label'>{asset.label}</span>
```
Comment (gavrilin-remote) [inline src/components/Dropdown/Dropdown.tsx:100]:
```suggestion
          <Typography className='customOption-label' text={option.label} />
```

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import Select, { Props, StylesConfig } from 'react-select';
+
+import { SelectArrow } from 'assets/svg';
+
+type Option = { label: string; value: string; image?: string };
```
Comment (gavrilin-remote) [inline src/components/Dropdown/Dropdown.tsx:7]: image can be SVGIcon

File: src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    <Select
+      components={{ DropdownIndicator: () => <SelectArrow /> }}
+      isSearchable={false}
+      options={props.options}
+      styles={styles}
+      {...props}
+      formatOptionLabel={(asset) => (
+        <div className='customOption'>
+          {asset.image && (
+            <img className='customOption-img' src={asset.image} alt='asset' />
```
Comment (gavrilin-remote) [inline src/components/Dropdown/Dropdown.tsx:98]: image can be SVGIcon

**Comment by marinakulsha** [comment]
Is it possible to maker short scroll?

File: src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (mr-teddy-bear) [reply src/components/Dropdown/Dropdown.tsx:95]: Done

File: src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (mr-teddy-bear) [reply src/components/Dropdown/Dropdown.tsx:100]: Done

File: src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (mr-teddy-bear) [reply src/components/Dropdown/Dropdown.tsx:98]: Done

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (mr-teddy-bear) [reply src/components/Dropdown/Dropdown.tsx:7]: Done

File: src/components/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
+import React, { FC } from 'react';
+
+import Select, { Props, StylesConfig } from 'react-select';
+
+import { SelectArrow } from 'assets/svg';
+import { Typography } from 'components';
+
+type Option = { label: string; value: string; picture?: FC };
```
Comment (gavrilin-remote) [inline src/components/Dropdown/Dropdown.tsx:8]:
```suggestion
type Option = { label: string; value: string; Icon?: SVGIcon };
```

File: src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  return (
+    <Select
+      components={{ DropdownIndicator: () => <SelectArrow /> }}
+      isSearchable={false}
+      options={props.options}
+      styles={styles}
+      {...props}
+      formatOptionLabel={({ picture, label }) => (
+        <div className='customOption'>
+          {picture && <div className='customOption-img'>{picture}</div>}
```
Comment (gavrilin-remote) [inline src/components/Dropdown/Dropdown.tsx:98]:
```suggestion
          {Icon && <Icon className='customOption-img' />}
```

File: src/components/Dropdown/Dropdown.tsx (hunk 5 above)
Comment (mr-teddy-bear) [reply src/components/Dropdown/Dropdown.tsx]: Done

File: src/components/Dropdown/Dropdown.tsx (hunk 6 above)
Comment (mr-teddy-bear) [reply src/components/Dropdown/Dropdown.tsx]: Done

**Review by gavrilin-remote — APPROVED** [review]

**Pull request merged**